common/persistence/sql/workflowStateMaps.go

```diff
@@ -483,25 +455,17 @@ func updateSignalInfos(
 			return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to execute update query. Error: %v", err))
 		}
 	}
-	if deleteInfo == nil {
-		return nil
-	}
-	result, err := tx.DeleteFromSignalInfoMaps(sqlplugin.SignalInfoMapsDeleteFilter{
-		ShardID:     shardID,
-		NamespaceID: namespaceID,
-		WorkflowID:  workflowID,
-		RunID:       runID,
-		InitiatedID: deleteInfo,
-	})
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to execute delete query. Error: %v", err))
-	}
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Failed to verify number of rows deleted. Error: %v", err))
-	}
-	if int(rowsAffected) != 1 {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update signal info. Deleted %v rows instead of 1", rowsAffected))
+
+	for _, deleteInfo := range deleteInfos {
+		if _, err := tx.DeleteFromSignalInfoMaps(sqlplugin.SignalInfoMapsDeleteFilter{
+			ShardID:     shardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  workflowID,
+			RunID:       runID,
+			InitiatedID: &deleteInfo,
```

Comment:
To be consistent you might want to rename to deleteInfoID, also InitiatedID does not need to be a pointer anymore.

Reply:
I'll approve the PR and you can land after renaming this.